contracts/pcv/utils/ERC20Allocator.sol

```diff
@@ -0,0 +1,459 @@
+        depositToEdit.decimalsNormalizer = decimalsNormalizer;
+        depositToEdit.targetBalance = targetBalance;
+
+        emit PSMUpdated(psm, token, targetBalance, decimalsNormalizer);
+    }
+
+    /// @notice disconnect an existing deposit from the allocator
+    /// @param psm Peg Stability Module to remove from allocation
+    function disconnectPSM(address psm) external override onlyGovernor {
+        delete allPSMs[psm];
+
+        emit PSMDeleted(psm);
+    }
+
+    /// @notice function to connect deposit to a PSM
+    /// this then allows the pulling of funds between the deposit and the PSM permissionlessly
+    /// as defined by the target balance set in allPSM's
+    /// this function does not check if the pcvDepositToPSM has already been connected
+    /// as only the governor can call and create, and overwriting with the same data (no op) is fine
+    /// @param psm peg stability module
+    /// @param pcvDeposit deposit to connect to psm
+    function connectDeposit(address psm, address pcvDeposit)
+        external
+        override
+        onlyGovernor
+    {
+        address pcvToken = allPSMs[psm].token;
+
+        /// assert pcv deposit and psm share same denomination
+        require(
+            PCVDeposit(pcvDeposit).balanceReportedIn() == pcvToken,
+            "ERC20Allocator: token mismatch"
+        );
+        require(pcvToken != address(0), "ERC20Allocator: invalid underlying");
+
+        pcvDepositToPSM[pcvDeposit] = psm;
+
+        emit DepositConnected(psm, pcvDeposit);
+    }
+
+    /// @notice delete an existing deposit
+    /// @param pcvDeposit PCV Deposit to remove connection to PSM
+    function deleteDeposit(address pcvDeposit) external override onlyGovernor {
+        delete pcvDepositToPSM[pcvDeposit];
+
+        emit DepositDeleted(pcvDeposit);
+    }
+
+    /// @notice sweep target token, this shouldn't ever be needed as this contract
+    /// does not hold tokens
+    /// @param token to sweep
+    /// @param to recipient
+    /// @param amount of token to be sent
+    function sweep(
+        address token,
+        address to,
+        uint256 amount
+    ) external onlyGovernor {
+        IERC20(token).safeTransfer(to, amount);
+    }
+
+    /// ----------- Permissionless PCV Allocation APIs -----------
+
+    /// @notice pull ERC20 tokens from PSM and send to PCV Deposit
+    /// if the amount of tokens held in the PSM is above
+    /// the target balance.
+    function skim(address pcvDeposit) external whenNotPaused {
+        address psm = pcvDepositToPSM[pcvDeposit];
+        _skim(psm, pcvDeposit);
+    }
+
+    /// @notice pull ERC20 tokens from PSM and send to PCV Deposit
+    /// if the amount of tokens held in the PSM is above
+    /// the target balance.
+    /// @param psm the peg stability to pull excess funds from
+    /// @param pcvDeposit the PCV Deposit to send excess funds to
+    function skim(address psm, address pcvDeposit) external whenNotPaused {
```

Comment:
we can remove this function as psm address can always be inferred from the pcvDeposit address